src/main/java/org/broadinstitute/barclay/help/BashTabCompletionDoclet.java

```diff
@@ -0,0 +1,356 @@
+        rootMap.put("tools", propertiesMap);
+
+        // Add the caller script options into another top-level tree node:
+        Map<String, Object> callerScriptOptionsMap = new HashMap<>();
+        callerScriptOptionsMap.put("callerScriptName", callerScriptName);
+
+        callerScriptOptionsMap.put("callerScriptPrefixLegalArgs", callerScriptPrefixLegalArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixArgValueTypes", callerScriptPrefixArgValueTypes);
+        callerScriptOptionsMap.put("callerScriptPrefixMutexArgs", callerScriptPrefixMutexArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixAliasArgs", callerScriptPrefixAliasArgs);
+        callerScriptOptionsMap.put("callerScriptPrefixMinOccurrences", callerScriptPrefixMinOccurrences);
+        callerScriptOptionsMap.put("callerScriptPrefixMaxOccurrences", callerScriptPrefixMaxOccurrences);
+        if ( hasCallerScriptPrefixArgs ) {
+            callerScriptOptionsMap.put("hasCallerScriptPrefixArgs", "true");
+        }
+        else {
+            callerScriptOptionsMap.put("hasCallerScriptPrefixArgs", "false");
+        }
+
+        callerScriptOptionsMap.put("callerScriptPostfixLegalArgs", callerScriptPostfixLegalArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixArgValueTypes", callerScriptPostfixArgValueTypes);
+        callerScriptOptionsMap.put("callerScriptPostfixMutexArgs", callerScriptPostfixMutexArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixAliasArgs", callerScriptPostfixAliasArgs);
+        callerScriptOptionsMap.put("callerScriptPostfixMinOccurrences", callerScriptPostfixMinOccurrences);
+        callerScriptOptionsMap.put("callerScriptPostfixMaxOccurrences", callerScriptPostfixMaxOccurrences);
+        if ( hasCallerScriptPostfixArgs ) {
+            callerScriptOptionsMap.put("hasCallerScriptPostfixArgs", "true");
+        }
+        else {
+            callerScriptOptionsMap.put("hasCallerScriptPostfixArgs", "false");
+        }
+
+        rootMap.put("callerScriptOptions", callerScriptOptionsMap);
+
+        // Get or create a template
+        final Template template = cfg.getTemplate(getIndexTemplateName());
+
+        // Create the output file
+        final File indexFile = new File(getDestinationDir(),
+                getIndexBaseFileName() + "." + getIndexFileExtension()
+        );
+
+        // Run the template and merge in the data
+        try (final FileOutputStream fileOutStream = new FileOutputStream(indexFile);
+             final OutputStreamWriter outWriter = new OutputStreamWriter(fileOutStream)) {
+            template.process(rootMap, outWriter);
+        } catch (TemplateException e) {
+            throw new DocException("Freemarker Template Exception during documentation index creation", e);
+        }
+    }
+
+    /**
+     * @return the name of the index template to be used for this doclet
+     */
+    @Override
+    public String getIndexTemplateName() { return "bash-completion.template.ftl"; }
```

Comment:
Nice that the extension is `flt` to show that it is a template. I think that we should do this for the other templates in Barclay...what do you think, @cmnbroad?